### ISSUE TITLE
Develop Basic Distributed-ML training Framework

### ISSUE THREAD

**Issue by msakarvadia** [msakarvadia/topology_aware_learning#1]
- [x] Get 2-tier FL working w/ logging
- [x] 2-teir FL: train models and graph training metrics
- [x] track accuracy
- [x] global test set loss + accuracy on clients + global model
- [x] Convert 2-teir set up into fully distributed by making a merged server client
- [x] Create example of many different distributed ML network topologies (including a 2 tiered one, hierarchical, etc.)
- [x] Make training graphs for distributed setting (think of some method to visualize information permeation through entire networks)
- [x] add adjacency matrix representation for topology (using networkX) -- [currently using ]
- [x] Simulate faulty network connections using predetermined probabilities
- [x] Deterministic out folder naming (so that we don't have to keep manually changing things)
- [x] #11
- [x] Make a class-wise accuracy testing function (https://stackoverflow.com/questions/39770376/scikit-learn-get-accuracy-scores-for-each-class): https://www.evidentlyai.com/classification-metrics/multi-class-metrics (sensitivity = recall = per class accuracy). Just log all scikitlearn metrics.
- [x] new fed split for data: https://github.com/h-flox/flox/blob/main/flight/learning/torch/utils.py#L175
